config/abilities/provider_any.rb

```diff
@@ -31,8 +31,12 @@
 
     if user.has_permission?(:partners)
       can [:show], AccountRelatedEvent do |event|
-        service_id = event.try(:service)&.id || event.try(:service_id)
-        !service_id || user.has_access_to_service?(service_id)
+        services = event.try(:services) || [event.try(:service)].compact
+        service_ids = services.map(&:id) || [event.try(:service_id)].compact
```

Comment:
@jlledom Have you seen any `AccountRelatedEvent` that was already including `service_ids`? 

On the other hand, you are now adding `services` property in `AccountRelatedEvent`, while this seems to be the only place where it is used. And if we only need the IDs, maybe this is what we can actually pass? `service_ids: [1,2,3]`, instead of `services: some_scope.to_a`

Reply:
> @jlledom Have you seen any `AccountRelatedEvent` that was already including `service_ids`?

No

> On the other hand, you are now adding `services` property in `AccountRelatedEvent`, while this seems to be the only place where it is used. And if we only need the IDs, maybe this is what we can actually pass? `service_ids: some_scope.pluck(:id)`, instead of `services: some_scope.to_a`

Yeah, I'll do that.